.github/workflows/nodejs-release.yml

```diff
@@ -1,56 +1,76 @@
-name: CI
+name: nodejs-release
 env:
   DEBUG: napi:*
-  APP_NAME: js-glaredb
-  MACOSX_DEPLOYMENT_TARGET: '10.13'
+  APP_NAME: "glaredb"
+  MACOSX_DEPLOYMENT_TARGET: "10.13"
+  PROTOC: "${{ github.workspace }}/deps/protoc/bin/protoc"
 permissions:
   contents: write
   id-token: write
-'on':
+on:
   push:
-    branches:
-      - main
-    tags-ignore:
-      - '**'
-    paths-ignore:
-      - '**/*.md'
-      - LICENSE
-      - '**/*.gitignore'
-      - .editorconfig
-      - docs/**
-  pull_request: null
+    tags:
+      - "*"
+  workflow_dispatch:
+defaults:
+  run:
+    working-directory: bindings/nodejs
+concurrency:
+  group: nodejs-release-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
   build:
     strategy:
-      fail-fast: false
+      fail-fast: true
       matrix:
         settings:
-          - host: macos-latest
+          - host: macos-12-xl
             target: x86_64-apple-darwin
             build: |
               yarn build
               strip -x *.node
-          - host: windows-latest
+          - host: windows-latest-8-cores
             build: yarn build
             target: x86_64-pc-windows-msvc
-          - host: ubuntu-latest
+          - host: ubuntu-2004-8-cores
```

Comment:
2004 seems pretty old? why do we pin to specific versions for macos and this?

Reply:
glibc

Since we're shipping a binary, we want as old as reasonably possible. 20.04 has 2.17 which works for a lot of environments.